### ISSUE TITLE
Ensure network protocol compatibility to taker `0.5.0` and `0.5.3`

### ISSUE THREAD

**Issue by da-kami** [itchysats/itchysats#2670]
It seems the API changes for support `ETHUSD` broke backwards compatibility in an unexpected way:

```
2022-08-09 14:12:29	
{"timestamp":"2022-08-09T14:12:29.904465676Z","level":"DEBUG","fields":{"message":"Failed to process maker offers: Failed to decode maker offers.: JSON error: missing field `leverage` at line 1 column 391: missing field `leverage` at line 1 column 391","peer":"12D3KooWEsK2X8Tp24XtyWh7DM65VfwXtNH2cmfs2JsWmkmwKbV1"},"target":"xtra_libp2p_offer::taker","span":{"err":"Failed to decode maker offers.","name":"fallible task handle_error"}}
	
2022-08-09 14:11:29	
{"timestamp":"2022-08-09T14:11:29.926697913Z","level":"DEBUG","fields":{"message":"Failed to process maker offers: Failed to decode maker offers.: JSON error: missing field `leverage` at line 1 column 391: missing field `leverage` at line 1 column 391","peer":"12D3KooWEsK2X8Tp24XtyWh7DM65VfwXtNH2cmfs2JsWmkmwKbV1"},"target":"xtra_libp2p_offer::taker","span":{"err":"Failed to decode maker offers.","name":"fallible task handle_error"}}
	
2022-08-09 14:10:29	
{"timestamp":"2022-08-09T14:10:29.86286764Z","level":"DEBUG","fields":{"message":"Failed to process maker offers: Failed to decode maker offers.: JSON error: unknown variant `EthUsd`, expected `BtcUsd` at line 1 column 76: unknown variant `EthUsd`, expected `BtcUsd` at line 1 column 76","peer":"12D3KooWEsK2X8Tp24XtyWh7DM65VfwXtNH2cmfs2JsWmkmwKbV1"},"target":"xtra_libp2p_offer::taker","span":{"err":"Failed to decode maker offers.","name":"fallible task handle_error"}}
	
2022-08-09 14:09:29	
{"timestamp":"2022-08-09T14:09:29.833906787Z","level":"DEBUG","fields":{"message":"Failed to process maker offers: Failed to decode maker offers.: JSON error: unknown variant `EthUsd`, expected `BtcUsd` at line 1 column 76: unknown variant `EthUsd`, expected `BtcUsd` at line 1 column 76","peer":"12D3KooWEsK2X8Tp24XtyWh7DM65VfwXtNH2cmfs2JsWmkmwKbV1"},"target":"xtra_libp2p_offer::taker","span":{"err":"Failed to decode maker offers.","name":"fallible task handle_error"}}
	
2022-08-09 14:08:59	
{"timestamp":"2022-08-09T14:08:59.833733485Z","level":"DEBUG","fields":{"message":"Failed to process maker offers: Failed to decode maker offers.: JSON error: missing field `leverage` at line 1 column 390: missing field `leverage` at line 1 column 390","peer":"12D3KooWEsK2X8Tp24XtyWh7DM65VfwXtNH2cmfs2JsWmkmwKbV1"},"target":"xtra_libp2p_offer::taker","span":{"err":"Failed to decode maker offers.","name":"fallible task handle_error"}}
```

Here's the [query](https://itchysats.grafana.net/explore?orgId=1&left=%7B%22datasource%22:%22grafanacloud-itchysats-logs%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22%7Bcontainer%3D~%5C%22autotaker_release%7Ctaker_release%5C%22,setup%3D%5C%22testnet%5C%22%7D%20!%3D%5C%22TRACE%5C%22%20%7C~%5C%22Failed%20to%20process%7CFailed%20to%20decode%5C%22%22%7D%5D,%22range%22:%7B%22from%22:%221660054084709%22,%22to%22:%221660054377004%22%7D%7D) to reprocude.

**Comment by luckysori** [comment]
I'm tackling this with #2680. The new protocol will be incompatible, but I will make sure that the old one doesn't suffer from these problems by:

- [x] Only allowing broadcasting offers of BTCUSD. See [this](https://github.com/itchysats/itchysats/blob/cc71f10d5177e80cd66cf26d7c28421aa5f95dd1/xtra-libp2p-offer/src/deprecated/protocol.rs#L52-L53).
- [x] Reintroducing the `leverage` field in the offer. See [this](https://github.com/itchysats/itchysats/blob/cc71f10d5177e80cd66cf26d7c28421aa5f95dd1/xtra-libp2p-offer/src/deprecated/protocol.rs#L93-L96).

**Comment by luckysori** [comment]
This should have been fixed in #2680, but let's monitor it.
